### DISCUSSION TITLE
Modeling site visits and observations of taxon occurrences

### DISCUSSION THREAD

**Discussion by dr-shorthair** [ternaustralia/ontology_tern#77] [Modelling]
Following up on a conversation with Kevin Thiele and @kheerand on 2021-12-08 regarding the O&M models for observing taxon occurrences during site visits/inventories, I prepared three Concept Maps for describing a notional instance which use models of increasing complexity.  Note that these are all 'valid' uses of the ontology, but have different levels of complexity supporting different levels of expressiveness.  

I find it easiest to explore what's going on with instance examples.  

## Observations of the Site

![Site inventory - simple](https://user-images.githubusercontent.com/608303/147041031-79cd89be-85e4-4e51-854e-4be4b96b6a4e.png)

In the **simple case** the site-as-a-whole is the feature-of-interest for observations made _in-situ_. The `observedProperty` is "Taxon Occurrence Present" with the observation _result_ being a specific `Taxon` instance. This property is commensurate with the type of the feature-of-interest - a Site. The procedure used for the (_in situ_) observation is "Field Inspection" and the sensor (observer) is a person denoted here by their ORCID. The details of the Taxon definitions are only shown in part - they would come from the NSL. 

This is the simplest representation following the [O&M](https://www.ogc.org/standards/om) model, as implemented in [SSN/SOSA](https://www.w3.org/TR/vocab-ssn/) which is the basis for the TERN Ontology. This interpretation elides the existence of the specific organisms (trees) at the site, which were judged to be a members of the taxa E. regnans and B. serrata.  The trees are not explicit in this description. The observation relates to the _site_ rather than an organism – the observed property is ‘taxon occurrence present’. This is the basic field-inventory of a site. However, this description does not allow for any sampling and separate confirmation of an identification in a museum or herbarium, etc.

The details of the observation description could be modelled a little differently, for example with the Taxon being a _parameter_ of the Observation, so the result would be ‘present’. But I was assuming that it is not normal to explicitly record ‘absent’ taxa so I just used the ‘result’ slot for the taxon. But this is really just implementation detail. The point is that it is the site that is being characterized through field observations. 

The set of observations made on the same site at the same time can be grouped in a `tern:SiteVisit` or alternatively in a [sosa:ObservationCollection](https://www.w3.org/TR/vocab-ssn-ext/#sosa:ObservationCollection). 

## _Ex-situ_ observations on samples taken from the site

![Site inventory - complex](https://user-images.githubusercontent.com/608303/147041063-3ba21f16-6a3f-45fc-9d76-ba4cb0626f59.png)

Stepping up one level of complexity, we might take a sample leaf from each identified tree, and then do the identification in a museum or herbarium. The observed property is now "Taxon Membership" which is commensurate with a different feature-of-interest type - here a leaf. The leaf is described as a Sample of the Site, and the Site is a sample of an Environmental System. The organism (tree) is still elided in this model. 

Under this scenario there is an act-of-sampling that takes place during the site-visit, and then an _ex situ_ act-of-observation later, at the museum. So the `resultTime` of the act of observation (i.e. identification) is later than the `phenomenonTime` of the result which is the sampling time, i.e. during the site-visit. Of course you don't have to have all this detail unless it is important. 

I showed the same person doing both sampling and observing, but that is not necessary. Under this scenario it is possible to check the original identification with another observation on the same leaf sample, which could use a different procedure, by a different observer, etc. 

## _In-situ_ observations at the site + _ex-situ_ observations on samples

![Site inventory - complete](https://user-images.githubusercontent.com/608303/147042046-acf5e8cc-344d-40bb-ada4-778da36617d9.png)

More complex still, in this scenario we recognise the tree as well. So the leaf is a sample of the tree which is a sample of the site which is a sample of the Environmental System. There are two acts-of-sampling shown - to spot the tree and to take the leaf - still **not shown** is the act-of-sampling which selected the Site!  The tree is at the site, so ST123 is an act of sampling to choose the one to identify and to take a leaf from. ST123 is probably not at all interesting - here it is shown primarily to justify the `isSampleOf` relation from the tree to the site. It can be inferred and probably need not be recorded in practice. 

This allows us to be more explicit about the two acts-of-observation - in the field it was on the tree, while in the museum it is on the leaf, with different results! The observed property is "Taxon Membership" again, which is commensurate with the feature-of-interest type - here a tree and a leaf. I haven't shown all the properties of every act or thing - you only need to record the ones that matter scientifically. All the samplings and the first observation are done by the same person, but a different person did the museum work.  

Note also that the individual things can be members of more than one class: the tree is a `Sample` in relation to the `Site`, and it is a `FeatureOfInterest` in relation to one act-of-sampling as well as one act-of-observation. 

The fact that there is a long chain of relations to get from the Taxon to the Site should not be a problem in querying a graph designed this way - [SPARQL supports property paths that can include ‘one or more’ matches of the same predicate](https://www.w3.org/TR/sparql11-query/#propertypaths).  This is where the `isSampleOf` links come in: you can query for an observation with a given `hasResult` where `hasFeatureOfInterest/isSampleOf*` ties it to a particular site of interest, or to find the sites where a particular taxon is observed.

**Comment by miekeGR** [comment]
Hi @dr-shorthair Thanks for sharing these examples, they are really helpful. Can I just clarify that these are O&M models, but not necessarily using the TERN specific ontology/vocabularies? I am still working through my understanding of Site in particular. When the Site is a Feature of Interest, what is the Feature Type of the Site using the TERN vocabularies?

**Comment by dr-shorthair** [comment]
The original conceptual model was defined in the [OGC/ISO specification for O&M](https://www.ogc.org/standards/om). 

The terminology in these diagrams is from the [SSN/SOSA implementation of O&M](https://www.w3.org/TR/vocab-ssn/) as an OWL ontology. 

The TERN Ontology makes some small additions to SOSA - `SiteVisit` and `EnvironmentalSystem` are from TERN, everything else is the shared SOSA/TERN terminology.

**Comment by dr-shorthair** [comment]
Updated the diagrams a little to show that Taxon instances are concepts that have (potentially multiple) names, and are not names directly. @kheerand

**Comment by F3RR3T** [comment]
@kheerand directed me here after our recent discussion on the quest for the "Ontology of Taxonomic Reference".

I'm interested in the use of `Taxon `as the range of an `Observation ` (i.e. Observation hasResult Taxon). I'm starting to think that we should explicitly use the class `TaxonConcept` rather than 'Taxon'. `TaxonConcept` has properties of `scientificName`, `vernacularName`, etc, but `Taxon` itself is almost a 'hidden' (unknowable?) class, namely 'that which is described by [zero, one or many] TaxonConcepts'. So the best we can do in assigning a result to an Observation is to assert a TaxonConcept, and record enough detail about the act of observation to bolster our claim that it is fact the 'correct' TaxonConcept. (i.e. that the organism belongs to the group that we have said it does).

If this abstraction of taxon makes sense, the act of 'naming' an organism (or group of them) is actually the application of a TaxonConcept; the taxon itself has no name. However in 'normal' usage, people will generally use 'taxon' to mean 'taxon concept'. (Reification)

**Reply by F3RR3T** [reply]
I should define how I'm using `TaxonConcept` before we go too much further. I'm aware that there is debate on this topic, but I have a definition in mind, which seems to work. I'll elaborate on this later.

**Reply by F3RR3T** [reply]
Consider the Ichthyologist that refused to eat oysters or prawns because they couldn't possibly be shellfish. A nice use of `TaxonConcept` would have broken the impasse.
